### PR TITLE
4.x: Helidon main class

### LINKED ISSUE #7075: Add `Nima` default main class, that bootstraps service registry and starts what should be started

### PR DESCRIPTION
**Pull request by tomas-langer**
Resolves #7075 

Introduction of `io.helidon.Main` extensible through service loader
Added CDI implementation 
Added Pico implementation

Modified MP app parent to use it
Refactored FT Example in Níma to use it

### CODE REVIEW

File: pico/api/src/main/java/io/helidon/pico/api/Helidon.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+    private static void basicInit() {
+        if (BASIC_INIT_DONE.compareAndSet(false, true)) {
+            LogConfig.configureRuntime();
+        }
+    }
+
+    private static final class ResettableHandler extends PicoServicesHolder {
+        @Deprecated
```
Comment (romain-grecourt) [inline pico/api/src/main/java/io/helidon/pico/api/Helidon.java:125]: Why `@Deprecated`, `ResettableHandler` is declared private.

File: pico/api/src/main/java/io/helidon/pico/api/Helidon.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    /**
+     * Initialize Helidon Injection service registry, and start all startable services.
+     */
+    public static void start() {
+        if (STARTED.compareAndSet(false, true)) {
+            basicInit();
+            registryInit(true);
+        } else {
+            LOGGER.log(System.Logger.Level.WARNING, "Duplicate call to method Helidon.start().");
```
Comment (romain-grecourt) [inline pico/api/src/main/java/io/helidon/pico/api/Helidon.java:76]: What about "Helidon.start() has already been called" ?

File: helidon/src/main/java/module-info.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Types need to start a Helidon application.
```
Comment (danielkec) [inline helidon/src/main/java/module-info.java:18]:
```suggestion
 * Types needed to start a Helidon application.
```

File: helidon/src/main/java/io/helidon/spi/package-info.java [hunk 4]
```diff
@@ -15,7 +15,7 @@
  */
 
 /**
- * Main entry point for Níma application.
- * Bootstraps required services, which implicitly starts server(s) available.
+ * Extension point to support a single main class ({@link io.helidon.Main}) to start various styles of applications,
+ * such as Helidon injection based applications, and Helidon MicroProfile applications.
```
[the remark this answers is not in the document]
Comment (tomas-langer) [reply helidon/src/main/java/io/helidon/spi/package-info.java:19]: yes

File: nima/common/pom.xml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
 
-    Copyright (c) 2022, 2023 Oracle and/or its affiliates.
+    Copyright (c) 2023 Oracle and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (tomas-langer) [reply nima/common/pom.xml:4]: mistake

File: pico/api/src/main/java/io/helidon/pico/api/Helidon.java (hunk 1 above)
Comment (tomas-langer) [reply pico/api/src/main/java/io/helidon/pico/api/Helidon.java:125]: because the super constructor is deprecated, and this will disable a warning

**Comment by tomas-langer** [comment]
Resolved all comments, needed to force push after rebasing.

**Review by danielkec — APPROVED** [review]

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**